app/resources/shop.py

```diff
@@ -41,22 +42,29 @@ def shop_info(data: dict, user: str):
 
 @m.user_endpoint(path=["shop", "buy"], requires=shop_buy)
 def shop_buy(data: dict, user: str):
-    product: str = data["product"]
+    products: dict = data["products"]
     wallet_uuid: str = data["wallet_uuid"]
     key: str = data["key"]
 
-    if product not in game_info["items"]:
-        return item_not_found
+    for product in products:
+        if product not in game_info["items"]:
+            return item_not_found
 
     if not exists_wallet(wallet_uuid):
         return wallet_not_found
 
-    price: int = game_info["items"][product]["price"]
+    boughtProducts: List[dict] = []
+
+    for product in products:
+        quantity: int = products[product]
+        price: int = game_info["items"][product]["price"] * quantity
+        response: dict = pay_shop(wallet_uuid, key, price, product)
```

Comment:
Don't buy each product seperately. Instead calculate the total price and create one transaction.